### PR TITLE
support iconPath for terminalProfile icon

### LINKED ISSUE #127607: Terminals/profiles/icon only supports codicon
Testing #127194

In the `package.json` I am only allowed to define an icon via codicons, not as path. We should as this aligns it with `commands/icon` and the vscode.d.ts definition of icons

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #127607

### CODE REVIEW

**Comment by meganrogge** [comment]
This currently doesn't work - icon is undefined (not properly passed through) somewhere

File: src/vs/workbench/contrib/terminal/browser/terminalIcon.ts [hunk 1]
```diff
@@ -6,19 +6,18 @@
...
-export function getColorClass(terminal: ITerminalInstance): string | undefined;
-export function getColorClass(terminalOrColorKey: ITerminalInstance | string): string | undefined {
+export function getColorClass(terminalOrColorKey: string | ITerminalInstance | ITerminalProfile | (ITerminalProfileContribution & { extensionIdentifier: string })): string | undefined {
 	let color = undefined;
 	if (typeof terminalOrColorKey === 'string') {
 		color = terminalOrColorKey;
-	} else if (terminalOrColorKey.color) {
-		color = terminalOrColorKey.color.replace(/\./g, '_');
+	} else if ('color' in terminalOrColorKey) {
+		color = terminalOrColorKey.color?.replace(/\./g, '_');
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/terminal/browser/terminalIcon.ts:20]: ignore this. It's for GHPR extension.

File: src/vs/workbench/contrib/terminal/browser/terminalIcon.ts [hunk 2]
```diff
@@ -6,19 +6,18 @@
 import { Codicon } from 'vs/base/common/codicons';
 import { hash } from 'vs/base/common/hash';
 import { URI } from 'vs/base/common/uri';
+import { ITerminalProfile } from 'vs/platform/terminal/common/terminal';
 import { ColorScheme } from 'vs/platform/theme/common/theme';
 import { ThemeIcon } from 'vs/platform/theme/common/themeService';
 import { ITerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { ITerminalProfileContribution } from 'vs/workbench/contrib/terminal/common/terminal';
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/terminal/browser/terminalIcon.ts:13]: ignore this. Testing GHPR extension.

File: src/vs/workbench/contrib/terminal/browser/terminalIcon.ts (hunk 1 above)
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/terminal/browser/terminalIcon.ts:20]: ignore this too. Testing GHPR extension.

**Pull request closed without merging**